cups/http.c

```diff
@@ -3916,6 +3916,7 @@ http_create(
   {
     _cupsSetError(IPP_STATUS_ERROR_INTERNAL, strerror(errno), 0);
     httpAddrFreeList(addrlist);
+    httpAddrFreeList(myaddrlist);
```

Comment:
I need to look at this more closely; my gut says we should just be freeing myaddlist and not addrlist (since, if supplied, we copy addrlist to myaddrlist...)